### PR TITLE
Add Tax Hold Type

### PR DESCRIPTION
**Pull request by ChristopherCotnoir**
Associated with xtuple/xtuple#3232, xtuple/private-extensions#994, and xtuple/xdruple-extension#54.

### CODE REVIEW

**Comment by mccjim** [comment]
So there have been requests over the years to add additional "Hold" types.  Did we add the ability to add 'Hold' types or did we just add the 1 'Tax Hold'?

**Comment by ChristopherCotnoir** [comment]
Just added the one, it is needed for xTupleCommerce.

File: common/storedProcErrorLookup.cpp [hunk 1]
```diff
@@ -1001,7 +1001,10 @@ const struct {
 			       "The Customer must return all materials for a "
 			       "related Return Authorization before any inven"
 			       "tory may be issued to this Order."),	0, "" },
-  { "issueToShipping", -15, QT_TRANSLATE_NOOP("storedProcErrorLookup", "The selected Sales Order is configured for Auto Registration. "
+  { "issueToShipping", -15, QT_TRANSLATE_NOOP("storedProcErrorLookup", "The selected Sales Order is on Tax Hold "
+                               "and must be taken off of Tax Hold before "
+                               "any inventory may be issued to it."),   0, "" },
+  { "issueToShipping", -16, QT_TRANSLATE_NOOP("storedProcErrorLookup", "The selected Sales Order is configured for Auto Registration. "
```
Comment (gilmoskowitz) [inline common/storedProcErrorLookup.cpp]: see comment on https://github.com/xtuple/xtuple/pull/3232

File: guiclient/openSalesOrders.cpp [hunk 2]
```diff
@@ -52,12 +52,14 @@ openSalesOrders::openSalesOrders(QWidget* parent, const char*, Qt::WindowFlags f
...
                             " UNION SELECT 4, '%5' "
+                            " UNION SELECT 5, '%6' "
                             " ORDER BY code; ")
           .arg(tr("None"))
           .arg(tr("Credit"))
           .arg(tr("Shipping"))
           .arg(tr("Packing"))
-          .arg(tr("Return"));
+          .arg(tr("Return"))
+          .arg(tr("Tax"));
```
Comment (gilmoskowitz) [inline guiclient/openSalesOrders.cpp]: Consider replacing this query with a series of `XComboBox::append()` calls. There's no need to trouble the database server to return a set of supplied constants.

Comment (ChristopherCotnoir) [reply guiclient/openSalesOrders.cpp]: Can't, it's used in a ParameterWidget combobox, and the function to add one of those only takes a query, not a preconstructed combobox.

**Pull request merged**